pyproject.toml

```diff
@@ -108,6 +109,7 @@ sphinx_rtd_theme = { version = "*", optional = true }
 autodoc_pydantic = { version = "*", optional = true }
 sphinx-reredirects = { version = "^0.1.2", optional = true }
 sphinx-automodapi = { version = "0.16.0", optional = true }
+groq = "^0.4.2"
```

Comment:
Shouldn't this be optional? We don't want a dependency on groq

Reply:
@someshfengde adding `optional = true` as done with the other dependencies should resolve this

Reply:
Hi @arnavsinghvi11  @okhat  I've added it as optional

Reply:
Also resolved the merge conflicts.

Reply:
Thanks @someshfengde ! are the poetry.lock changes needed for the PR as well? not sure if they are. will merge following this confirmation!

Reply:
👍